### PR TITLE
Trivial Usability Fixes

### PR DESCRIPTION
**Pull request by TheBlueMatt**
\+ panic = abort

### CODE REVIEW

File: src/bitcoind_client.rs [hunk 1]
```diff
@@ -67,7 +67,9 @@ impl BitcoindClient {
 		let http_endpoint = HttpEndpoint::for_host(host.clone()).with_port(port);
 		let rpc_credentials =
 			base64::encode(format!("{}:{}", rpc_user.clone(), rpc_password.clone()));
-		let bitcoind_rpc_client = RpcClient::new(&rpc_credentials, http_endpoint)?;
+		let mut bitcoind_rpc_client = RpcClient::new(&rpc_credentials, http_endpoint)?;
+		let _dummy = bitcoind_rpc_client.call_method::<BlockchainInfo>("getblockchaininfo", &vec![])
+			.await.expect("Failed to make initial call to bitcoind - please check your RPC user/password and access settings");
```
Comment (jkczyz) [inline src/bitcoind_client.rs:72]: Could you replace the `expect()` with `?` so this is handled by the caller?

File: src/main.rs [hunk 2]
```diff
@@ -290,6 +290,17 @@ async fn start_ldk() {
 		}
 	};
 
+	// Check that the bitcoin we've connected to is running the network we expect
```
Comment (jkczyz) [inline src/main.rs:293]: s/bitcoin/bitcoind

Unless this was a subtle dig at Charlie Munger. :P

File: src/main.rs [hunk 3]
```diff
@@ -290,6 +290,17 @@ async fn start_ldk() {
...
+	if bitcoind_client.get_blockchain_info().await.chain != match args.network {
+		bitcoin::Network::Bitcoin => "main",
+		bitcoin::Network::Testnet => "test",
+		bitcoin::Network::Regtest => "regtest",
+		bitcoin::Network::Signet => "signet",
+	} {
+		println!("Chain argument didn't match bitcoind chain");
```
Comment (valentinewallace) [inline src/main.rs:300]: Could you add in the LDK chain and the bitcoind chain into the log?

File: Cargo.toml [hunk 4]
```diff
@@ -8,35 +8,19 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-# lightning-background-processor = { git = "https://github.com/rust-bitcoin/rust-lightning", branch = "main" }
-lightning-background-processor = { git = "https://github.com/rust-bitcoin/rust-lightning", rev = "d4d322580994857b1222488f8467311d6db61482" }
-# lightning-background-processor = { path = "../rust-lightning/lightning-background-processor" }
+lightning = "0.0.14"
+lightning-block-sync = { version = "0.0.14", features = [ "rpc-client" ] }
```
Comment (valentinewallace) [inline Cargo.toml:12]: Thanks, forgot to remove all those comments

**Review by valentinewallace — APPROVED** [review]

**Review by valentinewallace — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Squashed. Net diff from first review is:
```
$ git diff-tree -U0 85ee4a4 HEAD
diff --git a/src/bitcoind_client.rs b/src/bitcoind_client.rs
index f374f6f..6d29ca0 100644
--- a/src/bitcoind_client.rs
+++ b/src/bitcoind_client.rs
@@ -71,2 +71,3 @@ impl BitcoindClient {
-		let _dummy = bitcoind_rpc_client.call_method::<BlockchainInfo>("getblockchaininfo", &vec![])
-			.await.expect("Failed to make initial call to bitcoind - please check your RPC user/password and access settings");
+		let _dummy = bitcoind_rpc_client.call_method::<BlockchainInfo>("getblockchaininfo", &vec![]).await
+			.map_err(|_| std::io::Error::new(std::io::ErrorKind::PermissionDenied,
+				"Failed to make initial call to bitcoind - please check your RPC user/password and access settings"))?;
diff --git a/src/cli.rs b/src/cli.rs
index c5dda5f..32119ae 100644
--- a/src/cli.rs
+++ b/src/cli.rs
@@ -106 +106,3 @@ pub(crate) async fn poll_for_user_input(
-	println!("LDK startup successful. To view available commands: \"help\".\nLDK logs are available at <your-supplied-ldk-data-dir-path>/.ldk/logs");
+	println!("LDK startup successful. To view available commands: \"help\".");
+	println!("LDK logs are available at <your-supplied-ldk-data-dir-path>/.ldk/logs");
+	println!("Local Node ID is {}.", channel_manager.get_our_node_id());
diff --git a/src/main.rs b/src/main.rs
index 8bf4293..7f4b5bf 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -293,2 +293,3 @@ async fn start_ldk() {
-	// Check that the bitcoin we've connected to is running the network we expect
-	if bitcoind_client.get_blockchain_info().await.chain != match args.network {
+	// Check that the bitcoind we've connected to is running the network we expect
+	let bitcoind_chain = bitcoind_client.get_blockchain_info().await.chain;
+	if bitcoind_chain != match args.network {
@@ -300 +301 @@ async fn start_ldk() {
-		println!("Chain argument didn't match bitcoind chain");
+		println!("Chain argument ({}) didn't match bitcoind chain ({})", args.network, bitcoind_chain);
$
```

File: src/main.rs [hunk 5]
```diff
@@ -292,13 +292,17 @@ async fn start_ldk() {
...
 	let bitcoind_chain = bitcoind_client.get_blockchain_info().await.chain;
-	if bitcoind_chain != match args.network {
-		bitcoin::Network::Bitcoin => "main",
-		bitcoin::Network::Testnet => "test",
-		bitcoin::Network::Regtest => "regtest",
-		bitcoin::Network::Signet => "signet",
-	} {
-		println!("Chain argument ({}) didn't match bitcoind chain ({})", args.network, bitcoind_chain);
+	if bitcoind_chain
+		!= match args.network {
```
Comment (jkczyz) [inline src/main.rs:296]: FWIW, it probably would be less braindead if you use a temporary variable for the network string. :P I tend to find combining `match` and `if` hard to read though admittedly it wasn't too bad here.

**Review by jkczyz — APPROVED** [review]

**Pull request merged**